ide/xml.xam/src/org/netbeans/modules/xml/xam/AbstractModel.java

```diff
@@ -179,12 +180,22 @@ public State getState() {
         return status;
     }
     
+    @Override
+    public String getStatusMessage() {
+        return statusMessage;
+    }
+    
     protected void setState(State s) {
+        setState(s, null);
+    }
+
+    protected void setState(State s, String message) {
```

Comment:
The `if` construct in line 193 (new) looks wrong - the state could stay the same, but the message could change.